source/common/http/filter/ip_tagging_filter.h

```diff
@@ -5,52 +5,94 @@
 #include <string>
 #include <vector>
 
+#include "envoy/common/exception.h"
+#include "envoy/config/filter/http/ip_tagging/v2/ip_tagging.pb.h"
 #include "envoy/http/filter.h"
-#include "envoy/json/json_object.h"
+#include "envoy/runtime/runtime.h"
 
-#include "common/common/assert.h"
-#include "common/json/config_schemas.h"
-#include "common/json/json_validator.h"
+#include "common/network/cidr_range.h"
+#include "common/network/lc_trie.h"
 
 namespace Envoy {
 namespace Http {
 
 /**
  * Type of requests the filter should apply to.
  */
-enum class FilterRequestType { Internal, External, Both };
+enum class FilterRequestType { INTERNAL, EXTERNAL, BOTH };
 
 /**
- * Configuration for the ip tagging filter.
+ * Configuration for the HTTP IP Tagging filter.
  */
-class IpTaggingFilterConfig : Json::Validator {
+class IpTaggingFilterConfig {
 public:
-  IpTaggingFilterConfig(const Json::Object& json_config)
-      : Json::Validator(json_config, Json::Schema::IP_TAGGING_HTTP_FILTER_SCHEMA),
-        request_type_(stringToType(json_config.getString("request_type", "both"))) {}
+  IpTaggingFilterConfig(const envoy::config::filter::http::ip_tagging::v2::IPTagging& config,
+                        const std::string& stat_prefix, Stats::Scope& scope,
+                        Runtime::Loader& runtime)
+      : request_type_(requestTypeEnum(config.request_type())), scope_(scope), runtime_(runtime),
+        stats_prefix_(stat_prefix + "ip_tagging.") {
+    if (config.ip_tags_size() == 0) {
```

Comment:
if this is true then why isn't it enforced in the proto?

Reply:
Required is removed in proto 3. Is there another way to enforce that it is specified?

Reply:
you can use protoc-gen-validate rules to enforce that the min number of items is 1 https://github.com/lyft/protoc-gen-validate#repeated

Reply:
i don't think we should necessarily enforce that the list has items though. what is the reason?

Reply:
I'll add the repeated tag in the proto. Once loading from a file system is supported, then letting the tags be empty would make sense. There would also have to be support around displaying the currently loaded ip tags.